Documentation/installation/k8s-install-migration.rst

```diff
@@ -0,0 +1,351 @@
+
+
+    At this point, you should have a cluster with Cilium installed and an overlay established, but no
+    pods managed by Cilium itself. You can verify this with the ``cilium`` command.
+
+    .. code-block:: shell-session
+
+      $ cilium status --wait
+      ...
+      Cluster Pods:     0/3 managed by Cilium
+
+
+6.  Create a :ref:`per-node config<per-node-configuration>` that will instruct Cilium to "take over" CNI networking
+    on the node. Initially, this will apply to no nodes; you will roll it out gradually via
+    the migration process.
+
+    .. code-block:: shell-session
+
+        cat <<EOF | kubectl apply --server-side -f -
+        apiVersion: cilium.io/v2alpha1
+        kind: CiliumNodeConfig
+        metadata:
+          namespace: kube-system
+          name: cilium-default
+        spec:
+          nodeSelector:
+            matchLabels:
+              io.cilium.migration/cilium-default: "true"
+          defaults:
+            write-cni-conf-when-ready: /host/etc/cni/net.d/05-cilium.conflist
+            custom-cni-conf: "false"
+            cni-chaining-mode: "none"
+            cni-exclusive: "true"
+        EOF
+
+Migration
+---------
+
+At this point, you are ready to begin the migration process. The basic flow is:
+
+Select a node to be migrated. It is not recommended to start with a control-plane node.
+
+.. code-block:: shell-session
+
+  $ NODE="kind-worker" # for the Kind example
+
+1.  Cordon and, optionally, drain the node in question.
+
+    .. code-block:: shell-session
+
+      $ kubectl cordon $NODE
+      $ kubectl drain --ignore-daemonsets $NODE
+
+    Draining is not strictly required, but it is recommended. Otherwise pods will encounter
+    a brief interruption while the node is rebooted.
+
+2.  Label the node. This causes the ``CiliumNodeConfig`` to apply to this node.
+
+    .. code-block:: shell-session
+
+      $ kubectl label node $NODE --overwrite "io.cilium.migration/cilium-default=true"
+
+3.  Restart Cilium. This will cause it to write its CNI configuration file.
+
+    .. code-block:: shell-session
+
+      $ kubectl -n kube-system delete pod --field-selector spec.nodeName=$NODE -l k8s-app=cilium
+      $ kubectl -n kube-system rollout status ds/cilium -w
+
+4.  Reboot the node.
+
+    If using kind, do so with docker:
+
+    .. code-block:: shell-session
+    
+      docker restart $NODE
+
+5.  Validate that the node has been successfully migrated.
+
+    .. code-block:: shell-session
+
+      $ cilium status --wait
+      $ kubectl get -o wide node $NODE
+      $ kubectl -n kube-system run --attach --rm --restart=Never verify-network \
+        --overrides='{"spec": {"nodeName": "'$NODE'", "tolerations": [{"operator": "Exists"}]}}' \
+        --image ghcr.io/nicolaka/netshoot:v0.8 -- /bin/bash -c 'ip -br addr && curl -s -k https://$KUBERNETES_SERVICE_HOST/healthz && echo'
+
+    Ensure the IP address of the pod is in the Cilium CIDR(s) supplied above.
```

Comment:
Ensure the IP address of the pod is in the Cilium CIDR(s) supplied above and that the apiservice is reachable.